contrib_bots/lib/tictactoe-bot.py

```diff
@@ -0,0 +1,345 @@
+                            "Here are the coordinates of each position. (Parentheses and spaces are optional). \n" \
+                            "(1, 1)  (1, 2)  (1, 3) \n(2, 1)  (2, 2)  (2, 3) \n(3, 1) (3, 2) (3, 3) \n"
+
+    def __init__(self, aboard):
+        self.board = aboard
+
+    def display_row(self, row):
+        ''' Takes the row passed in as a list and returns it as a string. '''
+        row_string = "[ "
+        for space in row:
+            row_string += space.strip() + " "
+        row_string += "]\n"
+        return row_string
+
+    def display_board(self, board):
+        ''' Takes the board as a nested list and returns a nice version for the user. '''
+        boardstring = ""
+        for row in board:
+            boardstring += output_mode(self.display_row(row), mode)
+        return boardstring
+
+    def board_is_full(self, board):
+        ''' Determines if the board is full or not. '''
+        full = False
+        board_state = ""
+        for row in board:
+            for space in row:
+                if space == "_":
+                    board_state += "_"
+        if "_" not in board_state:
+            full = True
+        return full
+
+    def win_conditions_2(self, board, triplets):
+        won = False
+        for set in triplets:
+            if board[set[0][0]][set[0][1]] == board[set[1][0]][set[1][1]] == board[set[2][0]][set[2][1]] != "_":
+                won = True
+        return won
+
+    def get_blank_locations(self, board):
+        ''' Gets the locations of the board that are still blank. Used when the computer makes its move. '''
+        blank_locations = []
+
+        for row in range(0, 3):
+            for col in range(0, 3):
+                if board[row][col] == "_":
+                    blank_locations.append([row, col])
+
+        return blank_locations
+
+    def get_x_locations(self, board):
+        ''' Gets the locations of the board that have x's placed in them by the user. This is mostly important when the
+        computer makes its first move, as some extra logic is implemented there. '''
+        full_locations = []
+        for row in range(0, 3):
+            for col in range(0, 3):
+                if board[row][col] == "x":
+                    full_locations.append([row, col])
+        return full_locations
+
+    def two_blanks(self, triplet, board):
+        ''' Determines which rows/columns/diagonals have two blank spaces and an 'o' already in them. It's more advantageous
+        for the computer to move there. This is used when the computer makes its move. '''
+        o_found = False
+        for position in triplet:
+            if board[position[0]][position[1]] == "o":
+                o_found = True
+                break
+
+        blanks_list = []
+        if o_found:
+            for position in triplet:
+                if board[position[0]][position[1]] == "_":
```

Comment:
This pattern is being repeated a lot. Maybe create a function which takes position and returns value?
```Python
def get_value(board, position):
    return board[position[0]][position[1]]
```